mg/agent.go

```diff
@@ -274,6 +278,9 @@ func NewAgent(cfg AgentConfig) (*Agent, error) {
 	return ag, nil
 }
 
+// Args returns a new instance of Args each time it is called. Please note that
+// the Store and Log are pointed to current Agent's properties and should not be
+// replaced.
 func (ag *Agent) Args() Args {
```

Comment:
I'd just document that it returns a copy of the agent's `Args`. There isn't anything in it for the caller to change anyway.